### ISSUE TITLE
Define api for specifying parameters for DropColumns and SelectColumns in AutoMLSearch

### ISSUE THREAD

**Issue by freddyaboulton** [alteryx/evalml#3150]
There are a couple of situations where AutoMLSearch will automatically add `DropColumns`/`SelectColumns` transformers to the set of pipelines it will search:

* We automatically drop index/unknown features in AutoMLSearch.
* In DefaultAlgorithm, we create a split pipeline for handling categorical features selected by the feature selector.
* For Known-in-Advance features (#3149), we create a split pipeline for handling known-in-advance features.

We need to make sure that the `columns` parameter of the `DropColumns`/`SelectColumns` components is properly set for all pipeline instances. In order to do so, we manually specify the parameters in multiple places:

* [_transform_parameters](https://github.com/alteryx/evalml/blob/main/evalml/automl/automl_algorithm/iterative_algorithm.py#L437) of `IterativeAlgorithm`
* [_create_split_select_parameters](https://github.com/alteryx/evalml/blob/main/evalml/automl/automl_algorithm/default_algorithm.py#L213) of `DefaultAlgorithm`
* [_set_additional_pipeline_params](https://github.com/alteryx/evalml/blob/main/evalml/automl/automl_algorithm/automl_algorithm.py#L137) of `AutoMLAlgorithm`

Our current approach is not ideal because it relies on hardcoding component names in multiple places and it sets the parameters in multiple places. The latter makes our implementation harder to understand and maintain. 

We should only have one way of specifying the parameters for select/drop columns transformers.

**Comment by chukarsten** [comment]
Agreed, this is a solid issue.  I'm going to add the spike tag to this as I think the first crack at it will be a design spike.

**Comment by freddyaboulton** [comment]
For more context on this issue, I think @jeremyliweishih and I have a thread in #3149 that I think captures the tech debt well:

https://github.com/alteryx/evalml/pull/3149#discussion_r768886264
